src/workflow/ArcanistBulkPatchWorkflow.php

```diff
@@ -0,0 +1,403 @@
+      $this->console->writeOut(
+        "<bg:red>** %s **</bg> %s\n",
+        pht('PATCH FAILURE'),
+        pht('Unable to find Git generated fallback patch.', $diffId));
+      return 1;
+    }
+
+    $passthru = new PhutilExecPassthru(
+      'git apply --3way --whitespace nowarn --index --verbose -- %s',
+      $patchfile);
+    $passthru->setCWD($this->getRepositoryAPI()->getPath());
+    $err = $passthru->execute();
+
+    if ($err) {
+      $this->console->writeOut(
+        "<bg:red>** %s **</bg> %s\n",
+        pht('PATCH FAILURE'),
+        pht('Unable to apply git generated patch - %s does not apply to current working state.', $diffId));
+    }
+    return $err;
+  }
+
+  // See ArcanistPatchWorkflow.php
+  private function getCommitMessage(ArcanistBundle $bundle) {
+    $revision_id    = $bundle->getRevisionID();
+    $commit_message = null;
+    $prompt_message = null;
+
+    // if we have a revision id the commit message is in differential
+
+    // TODO: See T848 for the authenticated stuff.
+    if ($revision_id && $this->isConduitAuthenticated()) {
+
+      $conduit        = $this->getConduit();
+      $commit_message = $conduit->callMethodSynchronous(
+        'differential.getcommitmessage',
+        array(
+          'revision_id' => $revision_id,
+        ));
+      $prompt_message = pht(
+        '  Note arcanist failed to load the commit message '.
+        'from differential for revision %s.',
+        "D{$revision_id}");
+    }
+
+    if (!$commit_message && $revision_id) {
+      $this->authenticateConduit();
+      $conduit        = $this->getConduit();
+      $commit_message = $conduit->callMethodSynchronous(
+        'differential.getcommitmessage',
+        array(
+          'revision_id' => $revision_id,
+        ));
+      $prompt_message = pht(
+        '  Note arcanist failed to load the commit message '.
+        'from differential for revision %s.',
+        "D{$revision_id}");
+    }
+
+    if (!$commit_message) {
+      throw new Exception(pht("Failed to fetch commit message."));
+    }
+
+    /* bulkpatch: Never ask for interactive commit message
+    // no revision id or failed to fetch commit message so get it from the
+    // user on the command line
+    if (!$commit_message) {
+      $template = sprintf(
+        "\n\n# %s%s\n",
+        pht(
+          'Enter a commit message for this patch. If you just want to apply '.
+          'the patch to the working copy without committing, re-run arc patch '.
+          'with the %s flag.',
+          '--nocommit'),
+        $prompt_message);
+
+      $commit_message = $this->newInteractiveEditor($template)
+        ->setName('arcanist-patch-commit-message')
+        ->editInteractively();
+
+      $commit_message = ArcanistCommentRemover::removeComments($commit_message);
+      if (!strlen(trim($commit_message))) {
+        throw new ArcanistUserAbortException();
+      }
+    }
+    */
+
+    return $commit_message;
+  }
+}
+// UBER CODE
```

Comment:
No need for this :) Whole file is Uber's :) We add these comments when we modify upstream files :)